File: 0605-can-place-flowers/0605-can-place-flowers.java
class Solution {
    public boolean canPlaceFlowers(int[] flowerbed, int n) {
        int empty = 1;

        for(int flower : flowerbed) {
            if(flower == 0) {
                empty++ ;
            }
            else  {
                n = n - ((empty - 1) / 2);
                empty = 0;
            }
        }
        n = n - (empty / 2);

        return n<=0;
    }
}

File: 0605-can-place-flowers/0605-can-place-flowers.py
class Solution(object):
    def canPlaceFlowers(self, flowerbed, n):
        """
        :type flowerbed: List[int]
        :type n: int
        :rtype: bool
        """
        place = [0] + flowerbed + [0]

        for i in range(1, len(place)-1) :
            if ( place[i-1] == 0 and place[i] == 0 and place[i+1] == 0 ) :
                place[i] = 1
                n -= 1

        return n <= 0